fenix/app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -279,6 +279,10 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
                     when (event?.action) {
                         MotionEvent.ACTION_DOWN, MotionEvent.ACTION_MOVE -> {
                             isPrivateButtonClicked = isTouchingPrivateButton(event.x, event.y)
+                            // Immediately drop Search Bar focus when the touch is not on the private button.
```

Comment:
@t-p-white nit around the comment: the code already tells us what the comment has described. What's missing is _why_ we do not clear focus if the private button is clicked.